src/charm.py

```diff
@@ -10,19 +10,24 @@
 https://discourse.charmhub.io/t/4208
 """
 import logging
-from typing import List
+from typing import List, Optional
 
+from charms.data_platform_libs.v0.s3 import (
+    S3Requirer,
+)
 from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
 from charms.loki_k8s.v0.loki_push_api import LokiPushApiConsumer
 from charms.mimir_coordinator_k8s.v0.mimir_cluster import MimirClusterProvider
 from charms.prometheus_k8s.v0.prometheus_remote_write import (
     PrometheusRemoteWriteConsumer,
 )
+from mimir_config import BUCKET_NAME, S3_RELATION_NAME, _S3ConfigData
```

Comment:
I think I said this before, but since it's a 'public' class (we're importing it here), we should remove the underscore prefix